openedx/features/edly/tests/test_middleware.py

```diff
@@ -71,3 +74,91 @@ def test_super_user_has_all_sites_access(self):
+        self.user = EdlyUserFactory()
+        self.request = RequestFactory().get('/')
+        self.request.user = self.user
+        self.request.site = SiteFactory()
+        EdlySubOrganizationFactory(lms_site=self.request.site)
+
+        self.client = Client(SERVER_NAME=self.request.site.domain)
+        self.client.cookies.load(
+            {
+                settings.EDLY_USER_INFO_COOKIE_NAME: cookies._get_edly_user_info_cookie_string(self.request)
+            }
+        )
+        self.client.login(username=self.user.username, password='test')
+
+    def test_settings_override_middleware_logs_warning_if_no_site_configuration_is_present(self):
+        """
+        Tests "SettingsOverrideMiddleware" logs warning if no site configuration is present.
+        """
+        with LogCapture(LOGGER_NAME) as logger:
+            self.client.get('/', follow=True)
+            logger.check_present(
+                (
+                    LOGGER_NAME,
+                    'WARNING',
+                    'Site ({site}) has no related site configuration.'.format(site=self.request.site)
+                )
+            )
+
+    def test_settings_override_middleware_logs_warning_if_site_configuration_has_no_values_set(self):
+        """
+        Tests "SettingsOverrideMiddleware" logs warning if site configuration has no values.
+        """
+        SiteConfigurationFactory(site=self.request.site)
+        with LogCapture(LOGGER_NAME) as logger:
+            self.client.get('/', follow=True)
+            logger.check_present(
+                (
+                    LOGGER_NAME,
+                    'WARNING',
+                    'Site configuration for site ({site}) has no values set.'.format(site=self.request.site)
+                )
+            )
+
+    def test_settings_override_middleware_logs_warning_if_site_configuration_has_no_django_settings_overrides(self):
+        """
+        Tests "SettingsOverrideMiddleware" logs warning if site configuration has no django settings override values.
+        """
+        SiteConfigurationFactory(
+            site=self.request.site,
+            values={
+                'PLATFORM_NAME': 'St. NormanX',
+            }
+        )
+        with LogCapture(LOGGER_NAME) as logger:
+            self.client.get('/', follow=True)
+            logger.check_present(
+                (
+                    LOGGER_NAME,
+                    'WARNING',
+                    'Site configuration for site ({site}) has no django settings overrides.'.format(site=self.request.site)
+                )
+            )
+
+    def test_settings_override_middleware_overrides_settings_correctly(self):
+        """
+        Tests "SettingsOverrideMiddleware" correctly overrides django settings.
+        """
+        SiteConfigurationFactory(
+            site=self.request.site,
+            values={
+                'DJANGO_SETTINGS_OVERRIDE': {
+                    'EDLY_USER_INFO_COOKIE_NAME': 'edly-user-info-test'
```

Comment:
We will never change this cookie name. Use some other variables which we will override in LMS, CMS (https://docs.google.com/spreadsheets/d/19smxeJAv4T3wr6KLfEXxkZP6_Ay0yiAFRETXVBRv9eQ/edit#gid=454576137)

Please use following list of variable to override (in same order)

`SITE_NAME`, 
`CMS_BASE`, 
`LMS_BASE`, 
`LMS_ROOT_URL`, 
`SESSION_COOKIE_DOMAIN`, 
`LOGIN_REDIRECT_WHITELIST`, 
`DEFAULT_FEEDBACK_EMAIL`,
`CREDENTIALS_PUBLIC_SERVICE_URL`

Reply:
done